src/UI/TransactionsTable/TransactionRow.tsx

```diff
@@ -0,0 +1,183 @@
+            globalStyles.dFlex,
+            globalStyles.alignItemsCenter
+          )}
+        >
+          <ExplorerLink
+            page={transaction.links.senderShardLink ?? ''}
+            data-testid='shardFromLink'
+          >
+            <ShardSpan shard={transaction.senderShard} />
+          </ExplorerLink>
+          <FontAwesomeIcon
+            icon={faArrowRight}
+            className={classNames(globalStyles.textSecondary, globalStyles.mx2)}
+          />
+          <ExplorerLink
+            page={transaction.links.receiverShardLink ?? ''}
+            data-testid='shardToLink'
+          >
+            <ShardSpan shard={transaction.receiverShard} />
+          </ExplorerLink>
+        </div>
+      </td>
+      <td>
+        <div
+          className={classNames(
+            globalStyles.dFlex,
+            globalStyles.alignItemsCenter
+          )}
+        >
+          {showLockedAccounts && (
+            <LockedTokenAddressIcon
+              address={transaction.sender}
+              tokenId={transaction.tokenIdentifier ?? ''}
+            />
+          )}
+          <ScAddressIcon initiator={transaction.sender} />
+          {directionOut ? (
+            <AccountName
+              address={transaction.sender}
+              assets={transaction.senderAssets}
+            />
+          ) : (
+            <>
+              {addressIsValid(transaction.sender) ? (
+                <ExplorerLink
+                  page={transaction.links.senderLink ?? ''}
+                  data-testid='senderLink'
+                  className={globalStyles.trimWrapper}
+                >
+                  <AccountName
+                    address={transaction.sender}
+                    assets={transaction.senderAssets}
+                  />
+                </ExplorerLink>
+              ) : (
+                <ShardSpan shard={transaction.sender} />
+              )}
+            </>
+          )}
+        </div>
+      </td>
+      {showDirectionCol && (
+        <td>
+          <div className={globalStyles.dFlex}>
+            <span
+              className={classNames(
+                styles.directionBadge,
+                transaction.transactionDetails.direction?.toLowerCase()
+              )}
+            >
+              {transaction.transactionDetails.direction?.toUpperCase()}
+            </span>
+          </div>
+        </td>
+      )}
+
+      <td>
+        <div
+          className={classNames(
+            globalStyles.dFlex,
+            globalStyles.alignItemsCenter
+          )}
+        >
+          {showLockedAccounts && (
+            <LockedTokenAddressIcon
+              address={receiver}
+              tokenId={transaction.tokenIdentifier ?? ''}
+            />
+          )}
+          <ScAddressIcon initiator={receiver} />
+          {directionIn ? (
+            <AccountName
+              address={receiver}
+              assets={transaction.receiverAssets}
+            />
+          ) : (
```

Comment:
there are many conditionals in TSX in this file. It would be cleaner to organise them in smaller components